lib/active_merchant/billing/gateways/flex_charge.rb

```diff
@@ -115,23 +135,29 @@ def add_invoice(post, money, credit_card, options)
           avsResultCode: options[:avs_result_code],
           cvvResultCode: options[:cvv_result_code],
           cavvResultCode: options[:cavv_result_code],
-          cardNotPresent: credit_card.verification_value.blank?
+          cardNotPresent: credit_card.is_a?(String) ? false : credit_card.verification_value.blank?
```

Comment:
Just confirming:  for TPV the cardNotPresent value will always be false.

But for credit cards, will it be something true if no verification value is provided?

Reply:
Flex charge Documentation is not clear on this topic but the messages during the testing led me to this conclusion.